app/src/main/java/com/example/cookbook/di/modules/AppModule.kt

```diff
@@ -0,0 +1,14 @@
+package com.example.cookbook.di.modules
+
+import com.example.cookbook.data.scheduler_provider.AppSchedulerProvider
+import com.example.cookbook.data.scheduler_provider.SchedulerProvider
+import dagger.Module
+import dagger.Provides
+import javax.inject.Singleton
+
+@Module
+class AppModule {
+    @Provides
+    @Singleton
+    fun provideSchedulerProvider(): SchedulerProvider = AppSchedulerProvider()
```

Comment:
You use SchedulerProvider  only for two Schedulers ?
I think we can get rid of it and just provide this two Schedulers in AppModule

Reply:
But if  in future you will use additional Schedulers  that make sense. Let's discuss it